docs/app/routes/docs/guides/react-three-fiber.mdx

```diff
@@ -0,0 +1,401 @@
+        color={springs.color}
+      />
+    </mesh>
+  )
+}
+
+export default function MyComponent() {
+  return (
+    <Canvas>
+      <ambientLight intensity={0.8} />
+      <pointLight intensity={1} position={[0, 6, 0]} />
+      <MyScene />
+    </Canvas>
+  )
+}
+```
+
+## Use the imperative API
+
+In the example below, we use the power of the `react-spring`'s api to lean into the imperative requirements
+of working with performant webGL scenes. The blob follows you round the canvas (lines 70-82) & scales on
+interaction (lines 46-56) without a single react render to cause these updates.
+
+In addition we use the provide a function to the `config` prop instead of an object to deliver a more sticky
+config for the spring in general, but to give the blob a bouncy feel when the `scale` key is changed – lines 18-30.
+
+Finally, because we're using the position of the mouse which can be considered a `Vector2` and the `position`
+of a `mesh` is a `Vector3` we use a custom interpolation via the `to` method of a `SpringValue` to interpolate
+the array, this can be seen on line 97.
+
+Why not read more about it the [imperative](/docs/concepts/imperative-api) and [interpolation](/docs/advanced/interpolation) APIs?
+
+```tsx live=true template=r3f lines=18-30,46-56,70-82,97 showLineNumbers defaultOpen=true
+import { useRef, useEffect, useCallback } from 'react'
+import { useSpring, animated } from '@react-spring/three'
+import { Canvas, useThree } from '@react-three/fiber'
+import { MeshDistortMaterial } from '@react-three/drei'
+
+const AnimatedMeshDistortMaterial = animated(MeshDistortMaterial)
+
+const MyScene = () => {
+  const isOver = useRef(false)
+
+  const { width, height } = useThree(state => state.size)
+
+  const [springs, api] = useSpring(
+    () => ({
+      scale: 1,
+      position: [0, 0],
+      color: '#ff6d6d',
+      config: key => {
+        switch (key) {
+          case 'scale':
+            return {
+              mass: 4,
+              friction: 10,
+            }
+          case 'position':
+            return { mass: 4, friction: 220 }
+          default:
+            return {}
+        }
+      },
+    }),
+    []
+  )
+
+  const handleClick = useCallback(() => {
+    let clicked = false
+
+    return () => {
+      clicked = !clicked
+      api.start({
+        color: clicked ? '#569AFF' : '#ff6d6d',
+      })
+    }
+  }, [])
+
+  const handlePointerEnter = () => {
+    api.start({
+      scale: 1.5,
+    })
+  }
+
+  const handlePointerLeave = () => {
+    api.start({
+      scale: 1,
+    })
+  }
+
+  const handleWindowPointerOver = useCallback(() => {
+    isOver.current = true
+  }, [])
+
+  const handleWindowPointerOut = useCallback(() => {
+    isOver.current = false
+
+    api.start({
+      position: [0, 0],
+    })
+  }, [])
+
+  const handlePointerMove = useCallback(
+    e => {
+      if (isOver.current) {
+        const x = (e.offsetX / width) * 2 - 1
+        const y = (e.offsetY / height) * -2 + 1
```

Comment:
I believe this NDC math is inbuilt into R3F's state model as `pointer`. I'd have to defer to @drcmda as for surrounding patterns but you could greatly reduce boilerplate with `<Canvas onCreated={(state) => state.events.connect(window)} />`.